db/python/layers/project_insights.py

```diff
@@ -301,367 +332,601 @@ async def _stats_es_indexes_query(
+            row["analysis_id"]: [
+                int(sgid) for sgid in row["sequencing_group_ids"].split(",")
+            ]
+            for row in _query_results
+        }
+        return sequencing_groups_by_analysis_id
+
+    async def get_analysis_sequencing_groups(self, all_group_analysis_rows: list[dict]):
         """
-        if not rows:
-            return 0
-        for row in rows:
-            if row['project'] == project and row['sequencing_type'] == sequencing_type:
-                return row[field]
-        return 0
-
-    @staticmethod
-    def get_val_for_project_and_sequencing_type_and_sg_id(
-        project, sequencing_type, sg_id, field, rows
-    ):
-        """
-        Filter returned records for a specific project, sequencing type and
-        sequencing group id and return the value of a specific field
+        Get the analysis IDs from the group analysis rows, which is a list of group analysis record dicts
         """
-        if not rows:
-            return 0
-        for row in rows:
-            if (
-                row['project'] == project
-                and row['sequencing_type'] == sequencing_type
-                and row['sequencing_group_id'] == sg_id
-            ):
-                return row[field]
-        return 0
-
-    @staticmethod
-    def get_val_for_project_and_sequencing_type_and_stage(
-        project, sequencing_type, stage, field, rows
+        analyses_to_query_sequencing_groups: list[int] = []
+        for group_analysis_rows in all_group_analysis_rows:
+            for row in group_analysis_rows.values():
+                analyses_to_query_sequencing_groups.append(row["id"])
+
+        return await self._get_sequencing_groups_by_analysis_ids(
+            analyses_to_query_sequencing_groups
+        )
+
+    def get_sg_web_report_links(
+        self,
+        sequencing_group_stripy_reports,
+        sequencing_group_mito_reports,
+        project,
+        sequencing_group_id: int,
     ):
         """
-        Filter returned records for a specific project, sequencing type and
-        stage and return the value of a specific field
+        Get the web report links for a sequencing group
         """
-        if not rows:
-            return 0
-        for row in rows:
-            if (
-                row['project'] == project
-                and row['sequencing_type'] == sequencing_type
-                and row['stage'] == stage
-            ):
-                return row[field]
-        return 0
-
-    async def get_project_insights_stats(
-        self, projects: list[int], sequencing_types: list[str]
+        report_links: defaultdict[str, dict[str, str]] = defaultdict(dict)
+        
+        stripy_report = sequencing_group_stripy_reports.get(
+            (project.id, sequencing_group_id)
+        )
+        if stripy_report:
+            report_links['stripy'] = {
+                'stripy_url': (
+                    f'https://main-web.populationgenomics.org.au/'
```

Comment:
Grab the gs path from the analysis-object, and add a transformation function. Easy to write tests for